arch/arc/core/fault.c

```diff
@@ -26,15 +26,24 @@
  * and is responsible only for reporting the error. Once reported, it then
  * invokes the user provided routine _SysFatalErrorHandler() which is
  * responsible for implementing the error handling policy.
- *
- * @return This function does not return.
  */
-void _Fault(const NANO_ESF *esf)
+void _Fault(NANO_ESF *esf)
 {
 	u32_t vector, code, parameter;
 	u32_t exc_addr = _arc_v2_aux_reg_read(_ARC_V2_EFA);
 	u32_t ecr = _arc_v2_aux_reg_read(_ARC_V2_ECR);
 
+#ifdef CONFIG_USERSPACE
+	/* TODO Improve this so that this logic only runs on MPU memory access
+	 * faults
+	 */
+	if (_current->fixup) {
```

Comment:
Are we sure that the fault handling in all of the userspace-supported platforms are not nested?